packages/backend/src/apps/toolbelt/actions/if-then/index.ts

```diff
@@ -0,0 +1,387 @@
+import { IGlobalVariable, IJSONArray, IJSONObject } from '@plumber/types'
+
+import defineAction from '@/helpers/define-action'
+import Step from '@/models/step'
+
+const MAX_BRANCHES = 50
+
+function evalCondition(parameters: IJSONObject): boolean {
+  const { field, is, condition, value } = parameters
+
+  let result: boolean
+  switch (condition) {
+    case 'equals':
+      result = field === value
+      break
+    case 'gte':
+      result = Number(field) >= Number(value)
+      break
+    case 'gt':
+      result = Number(field) > Number(value)
+      break
+    case 'lte':
+      result = Number(field) <= Number(value)
+      break
+    case 'lt':
+      result = Number(field) < Number(value)
+      break
+    case 'contains':
+      result = field.toString().includes(value.toString())
+      break
+    default:
+      throw new Error(
+        `Conditional logic block contains an unknown operator: ${condition}`,
+      )
+  }
+
+  if (is === 'not') {
+    result = !result
+  }
+
+  return result
+}
+
+function shouldTakeBranch($: IGlobalVariable): boolean {
+  const conditions = $.step.parameters.conditions as IJSONArray
+  return conditions.every((condition) =>
+    evalCondition(condition as IJSONObject),
+  )
+}
+
+/**
```

Comment:
Paiseh, can ignore this large block. I later realise the existence of a better approach (#190). But I leaving that approach as a separate PR because it's an RFC, and it relies on other mechanisms like `onPublish` hook.

In case team disagrees with #190, we still need this original comment block to explain our approach.